zvmsdk/hostops.py

```
@@ -38,6 +39,43 @@ def __init__(self):
         self._smtclient = smtclient.get_smtclient()
         self._volume_infos = {}
 
+    def _get_fcp_info(self):
+        _volumeop = volumeop.get_volumeop()
+
+        ret = _volumeop.get_all_fcp_usage_grouped_by_path()
```

Have you considered the case that no fcp_list configured? Any exception would be raised from this call?

---

@bjhuangr the UT case cover this.
```
        mock_get.return_value = {}
        ret = self._hostops._get_fcp_info()
        expected = {'total': 0, 'used': 0}
        self.assertEqual(expected, ret)
```

---

@SharpRazor  The UT cann't emulate what @bjhuangr is talking about. If I understand the code logic correctly, when no fcp_list configured, an SDKobjectnotexist exception will be raised from the db layer, instead of returning empty value.

---

thanks, done.